### PR TITLE
x and y facets different validation

### PR DESCRIPTION
**Pull request by BLAZEWIM**
# Pull Request

Fixes #[13](https://github.com/insightsengineering/teal.osprey/issues/13)

### CODE REVIEW

File: R/tm_g_spiderplot.R [hunk 1]
```diff
@@ -258,6 +258,14 @@ srv_g_spider <- function(id, data, filter_panel_api, reporter, dataname, label,
...
       validate(need(nrow(ADSL) > 0, "ADSL data has zero rows"))
       validate(need(nrow(ADTR) > 0, paste(dataname, "data has zero rows")))
+      if (length(xfacet_var) * length(yfacet_var) > 0) {
+        validate(
+          need(
+            !anyDuplicated(c(xfacet_var, yfacet_var)),
+            "X- and Y-facet variables must not be duplicated."
+          )
+        )
+      }
```
Comment (mhallal1) [inline R/tm_g_spiderplot.R:268]: is using `shinyvalidate` possible here?

**Comment by chlebowa** [comment]
Is it just me or is `shinyvalidate` quite bloaty?

**Comment by BLAZEWIM** [comment]
> Is it just me or is `shinyvalidate` quite bloaty?

Agree in many cases. You need to initialize class, enable it and then use validate anyway to trigger (+3 lines), also in this example you need to create another class inside the reactive. In many cases syntax can became quite unfriendly. From my perspective the only gain is putting warnings next to the fields.

**Comment by chlebowa** [comment]
> > Is it just me or is `shinyvalidate` quite bloaty?
> 
> Agree in many cases. You need to initialize class, enable it and then use validate anyway to trigger (+3 lines), also in this example you need to create another class inside the reactive. In many cases syntax can became quite unfriendly. From my perspective the only gain is putting warnings next to the fields.

Yeah, the fact that you have to build validators in reactives is additionally confusing. You can't even build them all in one place and then enable as necessary.

I do agree that highlighting the inputs is a good thing, ultimately, I'm just wondering whether there is a less verbose way of achieving that.

**Review by chlebowa — APPROVED** [review]

**Pull request merged**